lib/bolt/project.rb

```diff
@@ -130,36 +127,20 @@ def plans
       @data['plans']
     end
 
-    def project_directory_name?(name)
-      # it must match an installed project name according to forge validator
-      name =~ /^[a-z][a-z0-9_]*$/
-    end
-
-    def project_namespaced_name?(name)
-      # it must match the full project name according to forge validator
-      name =~ /^[a-zA-Z0-9]+[-][a-z][a-z0-9_]*$/
-    end
-
     def validate
-      n = @data['name']
-      if n && !project_directory_name?(n) && !project_namespaced_name?(n)
-        raise Bolt::ValidationError, <<~ERROR_STRING
-        Invalid project name '#{n}' in bolt-project.yaml; project names must match either:
-        An installed project name (ex. projectname) matching the expression /^[a-z][a-z0-9_]*$/ -or-
-        A namespaced project name (ex. author-projectname) matching the expression /^[a-zA-Z0-9]+[-][a-z][a-z0-9_]*$/
-        ERROR_STRING
-      elsif !project_directory_name?(name) && !project_namespaced_name?(name)
-        raise Bolt::ValidationError, <<~ERROR_STRING
-        Invalid project name '#{name}'; project names must match either:
-        A project name (ex. projectname) matching the expression /^[a-z][a-z0-9_]*$/ -or-
-        A namespaced project name (ex. author-projectname) matching the expression /^[a-zA-Z0-9]+[-][a-z][a-z0-9_]*$/
-
-        Configure project name in <project_dir>/bolt-project.yaml
-        ERROR_STRING
-      # If the project name is the same as one of the built-in modules raise a warning
-      elsif Dir.children(Bolt::PAL::BOLTLIB_PATH).include?(name)
-        raise Bolt::ValidationError, "The project '#{name}' will not be loaded. The project name conflicts "\
-          "with a built-in Bolt module of the same name."
+      if name
+        name_regex = /^[a-z][a-z0-9_]*$/
+        if name !~ name_regex
+          raise Bolt::ValidationError, <<~ERROR_STRING
```

Comment:
This is the error if I put `Bolt::Project.create_project(Dir.mktmpdir.to_s)` in `Bolt::Config.default`:

```
lucy ~/githubs/bolt (GH-1871-require-project-name) $ be bolt plan show
The project 'file' will not be loaded. The project name conflicts with a built-in Bolt module of the same name.
```

Reply:
This is interesting. It looks like the whole reason we need a default config is for the case where the config is invalid, since we use `config.format` and `config.color` to setup the outputter that we use to log the error. Because it's the default config, that means it disregards the output format the user wants. Maybe the right thing here is to not rely on the outputter at all for errors that happen this early in the lifecycle, so we can get rid of the default config entirely.

Separately, I do think it makes sense to have a `Bolt::Project.empty` or similar that will be used by the default config, rather than using the cwd. I will dig into that a bit more.